### PR TITLE
1.10/1563-ability-to-download-worker-list-with-a-click-of-a-button

### PR DESCRIPTION
**Pull request by AlinChindea**
### Trello card 👉 [https://trello.com/c/hs9uwODu/1563-ability-to-download-worker-list-with-a-click-of-a-button-ee-vv](https://trello.com/c/hs9uwODu/1563-ability-to-download-worker-list-with-a-click-of-a-button-ee-vv)

### CODE REVIEW

File: app/Http/Controllers/Service/Admin/CentreUsersController.php [hunk 1]
```diff
@@ -219,4 +219,45 @@ private function syncCentres($request, CentreUser $cu)
         // Sync them setting pivots.
         return $cu->centres()->sync($centre_ids);
     }
+
+    public function download()
+    {
+        $workers = CentreUser::get()->sortBy('homeCentre.name');
+        $csvExporter = new \Laracsv\Export();
```
Comment (charlesstrange2) [inline app/Http/Controllers/Service/Admin/CentreUsersController.php:226]: could be an import

File: app/Http/Controllers/Service/Admin/CentreUsersController.php [hunk 2]
```diff
@@ -219,4 +219,45 @@ private function syncCentres($request, CentreUser $cu)
...
+
+        /**
+         * * modify downloader values to print user friendly text
+         * * currently, the y/n in the model reverts to 0/1 because of Laravel casting
```
Comment (charlesstrange2) [inline app/Http/Controllers/Service/Admin/CentreUsersController.php:230]: 👍

File: resources/views/service/centreusers/index.blade.php [hunk 3]
```diff
@@ -59,6 +61,11 @@
                 @endforeach
             </tbody>
         </table>
+        <a href="{{ URL::route('admin.centreusers.download')}}" class="link" target="_blank">
```
Comment (charlesstrange2) [inline resources/views/service/centreusers/index.blade.php:64]: Check to see if they want it at the top or the bottom

File: routes/service.php [hunk 4]
```diff
@@ -34,100 +30,104 @@
...
         'uses' => 'Admin\DeliveriesController@index',
     ]);
     Route::get('deliveries/create', [
-        'as' =>'admin.deliveries.create',
+        'as' => 'admin.deliveries.create',
         'uses' => 'Admin\DeliveriesController@create',
     ]);
     Route::post('deliveries/store', [
-        'as' =>'admin.deliveries.store',
+        'as' => 'admin.deliveries.store',
```
Comment (charlesstrange2) [inline routes/service.php:130]: Loving the linting 👍

File: tests/Unit/Routes/ServiceRoutesTest.php [hunk 5]
```diff
@@ -22,6 +22,11 @@ class ServiceRoutesTest extends StoreTestCase
             'admin.centreusers.index' => [],
             'admin.centreusers.create' => [],
             'admin.centreusers.edit' => ['id' => 1],
+            /**
+             * * this route streams a csv file and that breaks the test as it tries to modify headers
```
Comment (charlesstrange2) [inline tests/Unit/Routes/ServiceRoutesTest.php:26]: 👍

File: resources/views/service/centreusers/index.blade.php (hunk 3 above)
Comment (AlinChindea) [reply resources/views/service/centreusers/index.blade.php:64]: dropped a messaged on the Trello card. will wait for them before merging as I realised it's also a minor case of copy - what should the button say and how should we title the file to be downloaded.

File: app/Http/Controllers/Service/Admin/CentreUsersController.php (hunk 1 above)
Comment (AlinChindea) [reply app/Http/Controllers/Service/Admin/CentreUsersController.php]: changed it to be an import

**Pull request merged**